sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java

```diff
@@ -189,14 +193,50 @@ public Context getSharedSpanBuilder(String spanName, Context context) {
         return context.addData(SPAN_BUILDER_KEY, getSpanBuilder(spanName, context));
     }
 
+    @Override
+    public void addEvent(String eventName, Map<String, Object> attributes, Instant timestamp) {
+        Span currentSpan = Span.current();
+        if (currentSpan == null) {
+            logger.info("Failed to find a starting span to associate an event with.");
+            return;
+        }
+
+        currentSpan.addEvent(
+            eventName,
+            attributes == null ? Attributes.empty() : convertToAttributes(attributes),
+            timestamp
+        );
+    }
+
+    /**
+     * Maps span/event properties to OpenTelemetry attributes.
+     *
+     * @param attributes the attributes provided by the client SDK's.
+     * @return the OpenTelemetry typed {@Link Attributes}.
+     */
+    private Attributes convertToAttributes(Map<String, Object> attributes) {
+        AttributesBuilder attributesBuilder = Attributes.builder();
+        attributes.forEach((key, value) -> {
+            if (value instanceof Boolean) {
+                attributesBuilder.put(key, (boolean) value);
+            } else if (value instanceof String) {
+                attributesBuilder.put(key, String.valueOf(value));
+            } else if (value instanceof Double) {
+                attributesBuilder.put(key, (Double) value);
+            } else if (value instanceof Long) {
+                attributesBuilder.put(key, (Long) value);
+            }
```

Comment:
What happens to other types like Float, Integer etc? Do we just drop those attributes?

Reply:
Currently, it does seem like the Otel's [AttributesBuilder](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/common/AttributesBuilder.java) doesn't support passing of float, int etc data types. So yes, any other data type would be ignored.